### PR TITLE
GH-75: using after channel instead of delay so exit is immediate

### PR DESCRIPTION
**Pull request by ataboo**
# Description

Using `time.After` channel instead of delay so that exit is immediate when application stops.

Fixes # GH-75

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] I have read the [contribution guidelines](https://github.com/pesos/grofer/blob/master/CONTRIBUTING.md) and followed it as far as possible. 
- [x] I have performed a self-review of my own code (if applicable)
- [x] I have commented my code, particularly in hard-to-understand areas (if applicable)
- [x] I have run `go fmt` on my code ([reference](https://blog.golang.org/gofmt))
- [x] I have made corresponding changes to the documentation (if applicable)
- [x] My changes generate no new warnings 
- [x] Any dependent and pending changes have been merged and published

### CODE REVIEW

**Comment by Gituser143** [comment]
@ataboo Thanks for the PR! This is a beautiful fix! But it needs to be implemented for other parts of the UI too. Take a look at other functions that serve data to the UI. A search for occurrences of `time.Sleep()` should point you at all the other functions.

**Comment by ataboo** [comment]
Okay, I'll have a look around for other sleep calls.

**Review by Gituser143 — APPROVED** [review]
This looks great! Unless you plan to make any more changes, I'll wait for a second review before merging this PR.

**Comment by ataboo** [comment]
Yeah I think that should cover this one.  Sounds good.

**Review by MadhavJivrajani — COMMENTED** [review]
Oh wow, that was really interesting, I learned something new today :p 

I had a couple of questions which I was hoping you could answer:
- Could the 'after' channel be created before the for loop and then used in the `select`? The reason I ask is since doing `time.After(d)` creates a new channel every iteration which will have to be subsequently cleaned up.
- Could `time.NewTimer()` be used instead of `time.After()` due to the reason mentioned [here](https://golang.org/pkg/time/#After)?

File: src/display/process/procGraphs.go
Comment (ataboo) [inline src/display/process/procGraphs.go:206]: This was more for me to spot if things were actually updating. It can come out if not needed.

**Comment by ataboo** [comment]
I think a ticker would fit the bill better if we're trying to prevent the performance hit for recreating the timer with After.

I consolidated the ticker with a callback for the action.  Note that I left a "last update" field in the individual proc display.  I think it'd be good for feedback and it was helpful for testing, but I can take it back out if you want.

**Comment by MadhavJivrajani** [comment]
> Note that I left a "last update" field in the individual proc display. I think it'd be good for feedback and it was helpful for testing, but I can take it back out if you want.

It's probably not needed in the display.

> I think a ticker would fit the bill better if we're trying to prevent the performance hit for recreating the timer with After.
I consolidated the ticker with a callback for the action.

Yes, that is a better fit for our issue. Thank you! I learnt quite a lot from your PR :smile:

**Review by MadhavJivrajani — APPROVED** [review]
:100:

**Comment by MadhavJivrajani** [comment]
Sorry about the revert commit reference, that was unintentional.

**Pull request merged**
